### DISCUSSION TITLE
Second monitor display black screen and mouse cursor

### DISCUSSION THREAD

**Discussion by khayo** [i3/i3#5950] [Q&A] [answered]
I recently installed Manjaro I3 and now I want to configure the system to use the second monitor

Apparently the monitor detection is ok, the mouse pointer appears on the screen, but the screen itself is blank.

If I use the command to open the terminal on the second monitor, the mouse cursor changes to the "text" cursor, but nothing appears on the screen. If I give a command (even without seeing what is being written), the results of the command happen in teem la. For example, if I use zenity to generate a notification popup and write this command in this "invisible terminal", the notification happens, this leads me to believe that the problem is in the terminal display, it exists but is not "rendered" . If I turn off the second monitor, the workspace that was created appears on the notebook screen, along with the terminal.

I've already used xrandr to do the configuration by hand:
xrandr --output DP-1 --auto --left-of eDP-1

I've already used arandr to configure the screen visually:
xrandr --output eDP-1 --primary --mode 1920x1080 --pos 1920x0 --rotate normal --output dp-1 --off --output dp-1 --off --output dp-3 --off --output HDMI-1 --mode 1920x1080 --pos 0x0 --rotate normal

I've already used autorandr to "automate" the settings.

A curious thing is that in certain configurations I make in arandr, the second screen starts to display an image and the first screen starts to have the behavior I mentioned previously.

I made the key settings in the configuration file:
```
set $note eDP-1
set $mon HDMI-1

#config second monitor
workspace 1 output $note
workspace 2 output $note
workspace 3 output $note
workspace 4 output $note
workspace 5 output $note
workspace 6 output $note
workspace 7 output $note
workspace 8 output $note
workspace 11 output $mon
workspace 12 output $mon
workspace 13 output $mon
workspace 14 output $mon

# Workspace names
# to display names or symbols instead of plain workspace numbers you can use
# something like: set $ws1 1:mail
#                 set $ws2 2:
set $ws1 1
set $ws2 2
set $ws3 3
set $ws4 4
set $ws5 5
set $ws6 6
set $ws7 7
set $ws8 8
set $ws11 11
set $ws12 12
set $ws13 13
set $ws14 14

# switch to workspace
bindsym $mod+1 workspace $ws1
bindsym $mod+2 workspace $ws2
bindsym $mod+3 workspace $ws3
bindsym $mod+4 workspace $ws4
bindsym $mod+5 workspace $ws5
bindsym $mod+6 workspace $ws6
bindsym $mod+7 workspace $ws7
bindsym $mod+8 workspace $ws8
bindsym Mod1+1 workspace $ws11
bindsym Mod1+2 workspace $ws12
bindsym Mod1+3 workspace $ws13
bindsym Mod1+4 workspace $ws14

# Move focused container to workspace
bindsym $mod+Ctrl+1 move container to workspace $ws1
bindsym $mod+Ctrl+2 move container to workspace $ws2
bindsym $mod+Ctrl+3 move container to workspace $ws3
bindsym $mod+Ctrl+4 move container to workspace $ws4
bindsym $mod+Ctrl+5 move container to workspace $ws5
bindsym $mod+Ctrl+6 move container to workspace $ws6
bindsym $mod+Ctrl+7 move container to workspace $ws7
bindsym $mod+Ctrl+8 move container to workspace $ws8
bindsym Mod1+Ctrl+1 move container to workspace $ws11
bindsym Mod1+Ctrl+2 move container to workspace $ws12
bindsym Mod1+Ctrl+3 move container to workspace $ws13
bindsym Mod1+Ctrl+4 move container to workspace $ws14

# Move to workspace with focused container
bindsym $mod+Shift+1 move container to workspace $ws1; workspace $ws1
bindsym $mod+Shift+2 move container to workspace $ws2; workspace $ws2
bindsym $mod+Shift+3 move container to workspace $ws3; workspace $ws3
bindsym $mod+Shift+4 move container to workspace $ws4; workspace $ws4
bindsym $mod+Shift+5 move container to workspace $ws5; workspace $ws5
bindsym $mod+Shift+6 move container to workspace $ws6; workspace $ws6
bindsym $mod+Shift+7 move container to workspace $ws7; workspace $ws7
bindsym $mod+Shift+8 move container to workspace $ws8; workspace $ws8
bindsym Mod1+Shift+1 move container to workspace $ws11; workspace $ws11
bindsym Mod1+Shift+2 move container to workspace $ws12; workspace $ws12
bindsym Mod1+Shift+3 move container to workspace $ws13; workspace $ws13
bindsym Mod1+Shift+4 move container to workspace $ws14; workspace $ws14
```

I've tried a lot of things, and nothing worked.

**Comment by khayo** [comment]
Additional Information:
- When I turn on the notebook and are asked for the password to decrypt the home partition, the message is shown on both monitors.
 
- Immediately upon system startup, both screens appear perfectly, including workspace 1 in the lower bar on the notebook screen and workspace 11 appears on the external monitor

- While I was writing this message, the notebook screen started to have the same behavior as the monitor, (black only with the mouse pointer), I removed the monitor cable and it came back.

- I use this monitor on another machine and it works perfectly. I removed the monitor from this other machine so that the automatic output selection would not cause confusion during testing.

**Reply by okraits** [reply]
From your description it's not ultimately clear in which order you are acting. You should:

- Connect the second monitor to your computer and turn it on
- Reconfigure your outputs with xrandr

Now i3 should use it and place workspaces on it, if you configured it to be an additional monitor and it's not just mirroring the first monitor.

Make sure that:

* xrandr, arandr and autorandr are not conflicting because of using all of them at the same time
* all your monitors are configured as separate outputs and with their native resolutions

> xrandr --output DP-1 --auto --left-of eDP-1

I would recommend to specify both outputs (both with `--auto`) if you want to use both of them.

**Reply by khayo** [reply]
I used this simplified format just to test, I really tested a lot (and learned a lot) until I found the solution, although I don't know exactly what this piece of code does to the system, it solved my problem

**Comment by khayo** [comment]
While I was doing more tests, my screen froze, I removed the monitor cable from the notebook, and put in a new one, it thawed the screen, after a while writing in a window, the second monitor started to show what it should. At the moment it is working, but probably when you turn it off and on it will stop working.t

**Comment by khayo** [comment]
as expected after removing the second monitor and putting it back in the problem returned

**Comment by stdstijn** [comment]
This won't help you in any way, but I have exactly the same situation/problem that you describe. Commenting for traction...

**Reply by khayo** [reply]
See my answer, I hope it helps you too

**Comment by khayo** [comment, marked as answer]
# TL!DR
Change the i3 configuration line where picom is called:
### Before
```bash
exec --no-startup-id nitrogen --restore; sleep 1; picom -b
```
### After
```bash
exec --no-startup-id nitrogen --restore; sganbialeep 1; seq 0 3 | xargs -l1 -I@ picom -b -d :0.@
```
# How did I find out how to solve the problem

I finally discovered the problem!
I noticed that if I insisted on turning the monitor on and off at some point it would work. Well, I started eliminating everything that had to do with image display. I killed conky, took the monitor off and put it back on. It didn't solve it, I did the same with the picom and this time the monitor worked correctly. I turned the monitor off and on several times and it worked.

I thought: I just turn off the picom when I put another monitor on, but it has the infamous habit of locking the system when it is closed. So this "solution" was discarded.
I continued looking for a solution to the problem until I came across the picom documentation on [archwiki] (https://wiki.archlinux.org/title/picom) and it is very straight to the point, to use multiple monitors it is necessary to start picom using this mystical language:
`seq 0 3 | xargs -l1 -I@ picom -b -d :0.@`

I really ***have no idea*** what I did when I ran this command, but it solved my problem.

The next step was to look in the i3 file where picom was started and change it to that startup, then I tested it and finally the problem was actually resolved. I tested installing a third monitor and everything worked perfectly. @stdstijn I hope this can help you solve your problem

**Reply by anotherglitchinthematrix** [reply]
My picom doesn't support -d flag, I'm using `vgit-89c2c`, can you tell me what version this flag is present?

edit 1:
Hmm, I think `DISPLAY=":0" picom -b` should be enough but do you think the DISPLAY variable is not set when the i3 starts? Also my problem is indeterministic, it's like 1 of 5 lottery.

edit 2:
I moved my picom exec after autorandr, so I hope it does make some change.

edit 3:
nothing changed, I also tried to follow the logs after bad state but there is nothing suspicios, I'm using very simple picom config, almost everything disabled, I'm using glx backend to get ridoff the hard edges of some interfaces, atleast I found the culprit, Thank you.
